### PR TITLE
zip/unzip files/folders

### PR DESCRIPTION
**Pull request by ivanvolkoff**
- Added option to zipping and unzipping folders/files.
- d.ts updated

### CODE REVIEW

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+    public static void zip(String path, String zipTargetPath) {
+        Path source = Paths.get(path);
+        Path target = Paths.get(zipTargetPath);
+        try {
+            zipFolder(source, target);
+            System.out.println("Done");
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (ThuF) [inline api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java:34]: Use logger to log the exception.

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+public class ZipProcessor {
+
+
+    public static void zip(String path, String zipTargetPath) {
+        Path source = Paths.get(path);
+        Path target = Paths.get(zipTargetPath);
+        try {
+            zipFolder(source, target);
+            System.out.println("Done");
```
Comment (ThuF) [inline api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java:32]: Remove the `System.out.println()`

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            e.printStackTrace();
+        }
+    }
+
+    public static void unzip(String zipPath, String targetPath) {
+        Path source = Paths.get(zipPath);
+        Path target = Paths.get(targetPath);
+        try {
+            unzipFolder(source, target);
+            System.out.println("Done");
```
Comment (ThuF) [inline api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java:43]: Same as above

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ThuF) [inline api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipProcessor.java:45]: Same as above

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipFacade.java [hunk 5]
```diff
@@ -19,17 +23,31 @@
...
  */
 public class ZipFacade {
-	
+
+	/**
+	 * Unzip zip archive from proivided zipPath to targetPath folder
+	 * @param zipPath
+	 * @param targetPath
+	 */
+	public static void importZip(String zipPath, String targetPath) {
```
Comment (ThuF) [inline api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipFacade.java:36]: Should we allow the `import/export` zip functionality only for files/folders under the `registry/public` path or for any path (as of now)?

**Review by ThuF — CHANGES_REQUESTED** [review]

File: api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipFacade.java (hunk 5 above)
Comment (ivanvolkoff) [reply api/api-facade/api-io/src/main/java/org/eclipse/dirigible/api/v3/io/ZipFacade.java:36]: in discussion with @vmutafov, we decided for now to allow any path. Of course, if there is no reason to limit yourself to "registry/public"

**Review by ThuF — APPROVED** [review]

**Pull request merged**